CMS: Update fetching-data for runtimeConfigPath, security scheme, and how security metadata works (#7876)
umbraco/UmbracoDocs: 17/umbraco-cms/customizing/foundation/fetching-data/custom-generated-client.md

}
```

> This uses the backoffice HTTP client directly for that request instead of the generated client. The `umbHttpClient` already has authentication and the correct base URL configured.

Reviewer: ```suggestion
The example above uses the backoffice HTTP client directly instead of the generated client. The `umbHttpClient` already has authentication and the correct base URL configured.
```

## How security metadata works

The `auth` callback on `umbHttpClient` only fires when a request carries `security` metadata. Here is how that metadata flows from your API to the generated client:

1. Each operation in your OpenAPI spec has an optional `security` field listing which security schemes it requires — for example, `[{ "Backoffice-User": [] }]`.
2. Those scheme names are defined in `components.securitySchemes`, where their type (OAuth2, HTTP Bearer, and so on) is declared.

> 3. When hey-api generates your client, it reads both and resolves them into its own runtime shape — `security: [{ type: 'http', scheme: 'bearer' }]` — emitted directly into each generated SDK function.

Reviewer: ```suggestion
1. Each operation in your OpenAPI spec has an optional `security` field listing the security schemes it requires. Example: `[{ "Backoffice-User": [] }]`.
2. Those scheme names are defined in `components.securitySchemes`, where their type (OAuth2, HTTP Bearer, and so on) is declared.
3. When hey-api generates your client, it reads both and resolves them into its own runtime shape - `security: [{ type: 'http', scheme: 'bearer' }]` - emitted directly into each generated SDK function.
```
